src/main/resources/additional_properties.json

```diff
@@ -18,5 +18,266 @@
         "$ref" : "#/$defs/serializer"
       }
     }
+  },
+  "processors" : {
+    "type" : "array",
+    "items" : {
+      "oneOf" : [
+        {
+          "type" : "object",
+          "required" : [ "cast_field" ],
+          "additionalProperties" : false,
+          "properties" : {
+            "cast_field" : {
+              "additionalProperties" : false,
+              "required" : [ "apply_to", "spec"],
+              "properties" : {
+                "apply_to" : {
+                  "$ref" : "#/$defs/apply_to"
+                },
+                "spec" : {
+                  "type" : "array",
+                  "items" : {
+                    "type": "string"
```

Comment:
That's a good point. We should discuss how we want to represent this in the UI and API.
```
"spec": [ "ID:string", "score:float64" ],
```
vs.
```
"spec": "ID:string,score:float64",
```
vs.
```
"spec": [ {"field": "ID", "type": string}, {"field": "score", "type": "float64"} ],
```
vs. (when no complex field)
```
"spec": "string",
```

Reply:
@ani-sha @uidoyen @indraraj what is your take on that one?